lib/common/utils.c

```diff
@@ -743,7 +743,7 @@ crm_pid_active(long pid, const char *daemon)
     } else if (rc == 0 && (daemon == NULL || have_proc_pid == -1)) {
         return 1;  /* kill as the only indicator, cannot double check */
 
-    } else {
+    } else if (daemon != NULL) {
```

Comment:
Good spot but semantically incorrect.
needs to be:
```diff
- } else if (rc < 0 && have_proc_pid == -1) {
+ } else if (rc < 0 && (have_proc_pid == -1 || daemon == NULL)) {
```
Please, fix master as well, or tell me.

Reply:
Also note that the only scenario applicable for the very unlikely
progress resulting in segfault is via `crm_procfs_pid_of` that is
just as well conceptially broken as with the recent CVE, but I wanted
to keep the scope at least partially limited, as the code makes
bad assumptions throughout...

Reply:
Thanks, if c18c3f1 from #1764 looks good, I'll backport it here

Reply:
Yes, please.  Thanks!

Reply:
Updated